spec/lib/item_marketplace/create_item_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'item_marketplace/create_item'
+require 'rails_helper'
+
+describe CreateItem do
+  context "item is created succesfully" do
+    let(:current_user) { User.create(username: "dreamteam", email: "us@dream.com", password_hash: "zzzzzzz") }
+    let(:item_params) { {name: "ping-pong paddle",
+                         description: "you beat Alex with it",
+                         buy_it_now_price: 1,
+                         seller_id: current_user.id} }
+    let(:tag_names)   { ["fun", "agile"] }
+    let(:create_item) { CreateItem.new(item_params, current_user, tag_names) }
+
+    it "returns the item" do
+
+      expect(create_item.call).to eq(Item.last)
+    end
+
+    it "it saves the item in the database" do
+      create_item.call
+
+      expect(Item.last.name).to eq("ping-pong paddle")
+      expect(Item.last.seller_id).to eq(current_user.id)
+    end
+
+    it "creates tag for the item" do
+      create_item.call
+
+      expect(Item.last.tags.count).to eq(2)
+      expect(Item.last.tags[0].name).to eq("fun")
+    end
+  end
+
+  context "item is not created succesfully" do
+    let(:current_user) { User.create(username: "dreamteam", email: "us@dream.com", password_hash: "zzzzzzz") }
+    let(:item_params) { {description: "you beat Alex with it",
+                         buy_it_now_price: 1,
+                         seller_id: current_user.id} }
+    let(:tag_names)   { ["fun", "agile"] }
+    let(:create_item) { CreateItem.new(item_params, current_user, tag_names) }
```

Comment:
Looks like you have 4 lets that have the same code except one key in one has. There's probably a better way to do that.